### DISCUSSION TITLE
Something went wrong installing the "sharp" module

### DISCUSSION THREAD

**Discussion by xzg5101** [donmccurdy/glTF-Transform#1049] [Q&A] [answered]
Environment：Windows 11, Node v18.17.0, Typescript, uses Yarn for other packages except for `glTF-Transform`.
When I add following line calling `glTF-Transform` to the project and run `yarn start` or `npm start`:
![image](https://github.com/donmccurdy/glTF-Transform/assets/72267123/0c33113d-367c-4998-b61f-4f7b3db5fcb5)
I recieve following error:
```
D:\path\to\repo\node_modules\sharp\lib\sharp.js:37
  throw new Error(help.join('\n'));
        ^
Error:
Something went wrong installing the "sharp" module

The specified procedure could not be found.
\\?\D:\path\to\repo\node_modules\sharp\build\Release\sharp-win32-x64.node

Possible solutions:
- Install with verbose logging and look for errors: "npm install --ignore-scripts=false --foreground-scripts --verbose sharp"
- Install for the current win32-x64 runtime: "npm install --platform=win32 --arch=x64 sharp"
- Consult the installation documentation: https://sharp.pixelplumbing.com/install
    at Object.<anonymous> (D:\path\to\repo\node_modules\sharp\lib\sharp.js:37:9)      
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Function.Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (D:\path\to\repo\node_modules\sharp\lib\constructor.js:11:1)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
But if I comment out that line, the issue disappears. 

Here is what I have tried so far:
1.  Install with verbose logging and look for errors: "npm install --ignore-scripts=false --foreground-scripts --verbose sharp"
The installation was successful, but the issue presists.
2. Install for the current win32-x64 runtime: "npm install --platform=win32 --arch=x64 sharp"
The installation was successful, but the issue presists.
3. Check the file: ```D:\path\to\repo\node_modules\sharp\build\Release\sharp-win32-x64.node```
The file does exist.
4. Remove `node_modules\sharp` and reinstall sharp || Remove `node_modules` and reinstall all packages
The installation was successful, but the issue presists.
5. Update Node
My node version is v18.17.0
6. Clone the repository in a new directory
Issue presists.
7. I am not allowed to upload the project and I failed to reproduce it in a new repo. I tried to install sharp first in a empty yarn project and then add glTF-Transform, finally call both in index.js, but there was no error reported and glTF-Transform worked flawlessly as expected.

```
The specified procedure could not be found.
\\?\D:\path\to\repo\node_modules\sharp\build\Release\sharp-win32-x64.node
```
The file was required as follow in `node_modules\sharp\lib\sharp.js:37`:
```
try {
  module.exports = require(`../build/Release/sharp-${platformAndArch}.node`);
} catch (err) {
...
}
```

Any potential fix, thoughts or suggestion is appreciated.

**Comment by donmccurdy** [comment, marked as answer]
Hi @xzg5101! I have no control over the "sharp" module, but there are some troubleshooting tips at https://sharp.pixelplumbing.com/install ... if you are located in China by any chance, you may need to follow the "Chinese Mirror" section of that guide.

**Reply by xzg5101** [reply]
Thank you so much! I followed the instruction in the link and removed canvas; now everything works! The issue is due to the conflict betweem sharp and canvas. I actually saw the page you sent me but sadly I wasn't patient enough to slide all the way to the very bottom : (

**Comment by jameszhong2008** [comment]
Hi @donmccurdy, I try install @gltf-transform/cli to compress *.glb model. 
refer to: [install gltf-transform](https://discourse.threejs.org/t/how-to-load-a-gltf-that-uses-ktx2-textures-getting-error-setktx2loader-must-be-called-before-loading-ktx2-textures/48792/6)
when I run command:

`gltf-transform etc1s model.glb compressed.glb`

Got following error.

```
C:\Users\Admin\AppData\Roaming\nvm\v20.2.0\node_modules\@gltf-transform\cli\node_modules\sharp\lib\sharp.js:37
  throw new Error(help.join('\n'));
        ^

Error:
Something went wrong installing the "sharp" module

Cannot find module '../build/Release/sharp-win32-x64.node'
Require stack:
- C:\Users\Admin\AppData\Roaming\nvm\v20.2.0\node_modules\@gltf-transform\cli\node_modules\sharp\lib\sharp.js
- C:\Users\Admin\AppData\Roaming\nvm\v20.2.0\node_modules\@gltf-transform\cli\node_modules\sharp\lib\constructor.js
- C:\Users\Admin\AppData\Roaming\nvm\v20.2.0\node_modules\@gltf-transform\cli\node_modules\sharp\lib\index.js
```

I search on web, this issue may cause by canvas installed. But I didn't install canvas, I don't know how to check and solve it.

Any suggestion is appreciated.

Thanks.

**Reply by donmccurdy** [reply]
Please try the advice in my answer above – does that link work for you?

**Reply by jameszhong2008** [reply]
Thanks a lot, sharp was not installed correctly caused this problem.
